lib/hotbar.js

```diff
@@ -72,72 +180,270 @@ class HotBar extends LitElement {
+    .hotbar-item canvas {
+      width: 100%;
+      height: 100%;
     }
 
-    #hotbar-item-name {
+    .slot-stack {
+      position: absolute;
+      right: calc(1px * var(--guiScale));
+      bottom: calc(1px * var(--guiScale));
+
+      font-family: mojangles, minecraft, monospace;
+      font-size: calc(10px * var(--guiScale));
+
       color: white;
+      text-shadow: calc(1px * var(--guiScale)) calc(1px * var(--guiScale)) black;
+    }
+
+    .hotbar-slot.selected::after {
+      content: '';
+      display: block;
+      position: absolute;
+      top: calc(-1px * var(--guiScale));
+      left: calc(-2px * var(--guiScale));
+    
+      width: calc(24px * var(--guiScale));
+      height: calc(22px * var(--guiScale));
+    
+      background-image: url('textures/1.16.4/gui/widgets.png');
+      background-size: calc(256px * var(--guiScale));
+      background-position-y: calc(-22px * var(--guiScale));
+    }
+
+    .experience-bar {
       position: absolute;
-      top: 100%;
+      bottom: calc(24px * var(--guiScale));
       left: 50%;
-      transform: translate(-50%, calc(-170px));
-      margin-top: 0px;
-      text-shadow: rgb(63, 63, 63) 2px 2px 0px;
+      transform: translate(-50%);
+
+      width: calc(182px * var(--guiScale));
+      height: calc(5px * var(--guiScale));
+    }
+
+    .experience-bar-progress {
+      position: absolute;
+      bottom: 0;
+      left: calc(50% - (91px * var(--guiScale)));
+
+      width: 0%;
+      height: 100%;
+    }
+
+    .experience-bar-progress.empty {
+      width: 100%;
+      background-image: url('extra-textures/icons.png');
+      background-size: calc(256px * var(--guiScale));
+      background-position-y: calc(-64px * var(--guiScale));
+    }
+
+    .experience-bar-progress.full {
+      background-image: url('extra-textures/icons.png');
+      background-size: calc(256px * var(--guiScale));
+      background-position-y: calc(-69px * var(--guiScale));
+    }
+
+    .experience-label {
+      position: absolute;
+      bottom: calc(1px * var(--guiScale));
+      left: 50%;
+      transform: translate(-50%);
+
       font-family: mojangles, minecraft, monospace;
-      font-size: 24px;
-      text-align: center;
+      font-size: calc(10px * var(--guiScale));
+
+      color: rgb(128, 255, 32);
+      z-index: 1;
     }
 
-    .hotbar-item-name-fader {
-      opacity: 0;
-      transition: visibility 0s, opacity 2s linear;
-      transition-delay: 2s;
+    .experience-label.shadow1 {
+      color: black;
+      left: calc(50% - (1px * var(--guiScale)));
+    }
+
+    .experience-label.shadow2 {
+      color: black;
+      left: calc(50% + (1px * var(--guiScale)));
+    }
+
+    .experience-label.shadow3 {
+      color: black;
+      bottom: calc(2px * var(--guiScale));
+    }
+
+    .experience-label.shadow4 {
+      color: black;
+      bottom: calc(0px * var(--guiScale));
+    }
+
+    .heart-renderer {
+      position: absolute;
+      bottom: calc(30px * var(--guiScale));
```

Comment:
That's a huge amount of css
Did you consider using css grid instead of calc ?

Reply:
IMO every elements should be set to a fixed scale and the whole hotbar should be scaled using css transform, that would reduce the code a lot.